wrappers/mongo/mongo.go

```diff
@@ -0,0 +1,504 @@
+		logOperationFailure("Could not complete readCursor", err.Error())
+	}
+
+	marshalToMetadata(event.Resource.Metadata, "filter", args[1])
+	marshalToMetadata(event.Resource.Metadata, "documents", docs)
+	completeMongoEvent(event)
+	return response
+}
+
+
+func (coll *MongoCollectionWrapper) FindOne(args ...interface{}) interface{} {
+	event := startMongoEvent(currentFuncName(), coll)
+	response := coll.collection.FindOne(
+		args[0].(context.Context),
+		args[1].(interface{}),
+	)
+
+	var document map[string]string
+	decodedResponse := reflect.ValueOf(response).
+		MethodByName("Decode").
+		Call([]reflect.Value{ reflect.ValueOf(&document) })
+	if err := decodedResponse[0]; err.Interface() != nil {
+		logOperationFailure("Could not complete Decode SingleResult", err.String())
+		epsagon.ExtractTracer([]context.Context{}).AddExceptionTypeAndMessage(
+			"mongo-driver",
+			err.String(),
+		)
+	}
+
+	marshalToMetadata(event.Resource.Metadata, "params", args[1])
+	marshalToMetadata(event.Resource.Metadata, "document", document)
+	completeMongoEvent(event)
+	return response
+}
+
+
+func (coll *MongoCollectionWrapper) FindOneAndDelete(args ...interface{}) interface{} {
+	event := startMongoEvent(currentFuncName(), coll)
+	response := coll.collection.FindOneAndDelete(
+		args[0].(context.Context),
+		args[1].(interface{}),
+	)
+
+	var document map[string]string
+	decodedResponse := reflect.ValueOf(response).
+		MethodByName("Decode").
+		Call([]reflect.Value{ reflect.ValueOf(&document) })
+	if err := decodedResponse[0]; err.Interface() != nil {
+		logOperationFailure("Could not complete Decode SingleResult", err.String())
+		epsagon.ExtractTracer([]context.Context{}).AddExceptionTypeAndMessage(
+			"mongo-driver",
+			err.String(),
+		)
+	}
+
+	marshalToMetadata(event.Resource.Metadata, "params", args[1])
+	marshalToMetadata(event.Resource.Metadata, "document", document)
+	completeMongoEvent(event)
+	return response
+}
+
+
+func (coll *MongoCollectionWrapper) FindOneAndReplace(args ...interface{}) interface{} {
+	event := startMongoEvent(currentFuncName(), coll)
+	response := coll.collection.FindOneAndReplace(
+		args[0].(context.Context),
+		args[1].(interface{}),
+		args[2].(interface{}),
+	)
+	var document map[string]string
+	decodedResponse := reflect.ValueOf(response).
+		MethodByName("Decode").
+		Call([]reflect.Value{ reflect.ValueOf(&document) })
+	if err := decodedResponse[0]; err.Interface() != nil {
+		logOperationFailure("Could not complete Decode SingleResult", err.String())
+		epsagon.ExtractTracer([]context.Context{}).AddExceptionTypeAndMessage(
+			"mongo-driver",
+			err.String(),
+		)
+	}
+
+	marshalToMetadata(event.Resource.Metadata, "filter", args[1])
+	marshalToMetadata(event.Resource.Metadata, "replacement", args[2])
+	marshalToMetadata(event.Resource.Metadata, "document", document)
+	completeMongoEvent(event)
+	return response
+
+}
+
+
+func (coll *MongoCollectionWrapper) FindOneAndUpdate(args ...interface{}) interface{} {
```

Comment:
I think the wrappers should match the original functions signatures (and not receive args ... interface{})
Let's talk about it :)